dotnet/src/SemanticKernel.UnitTests/Contents/ImageContentTests.cs

```diff
@@ -26,4 +28,107 @@ public void ToStringReturnsString()
+    {
+        // Arrange
+        var data = BinaryData.FromString("this is a test", "text/plain");
+        var content1 = new ImageContent(data);
+
+        // Act
+        var result1 = content1.ToString();
+        var dataUriToExpect = $"data:{data.MediaType};base64,{Convert.ToBase64String(data.ToArray())}";
+
+        // Assert
+        Assert.Equal(dataUriToExpect, result1);
+    }
+
+    [Fact]
+    public void ToStringForUriAndDataUriReturnsDataUriString()
+    {
+        // Arrange
+        var data = BinaryData.FromString("this is a test", "text/plain");
+        var content1 = new ImageContent(data);
+        content1.Uri = new Uri("https://endpoint/");
+
+        // Act
+        var result1 = content1.ToString();
+        var dataUriToExpect = $"data:{data.MediaType};base64,{Convert.ToBase64String(data.ToArray())}";
+
+        // Assert
+        Assert.Equal(dataUriToExpect, result1);
+    }
+
+    [Fact]
+    public void CreateForNullDataUriOrWithoutMediaTypeThrows()
+    {
+        // Arrange
+        var data = BinaryData.FromString("this is a test", null);
+        var data2 = BinaryData.FromString("this is a test", string.Empty);
+        var data3 = BinaryData.FromString("this is a test", " ");
+
+        // Assert
+        Assert.Throws<ArgumentNullException>(() => new ImageContent((BinaryData)null!));
+        Assert.Throws<ArgumentNullException>(() => new ImageContent(data));
+        Assert.Throws<ArgumentNullException>(() => new ImageContent(data2));
+        Assert.Throws<ArgumentNullException>(() => new ImageContent(data3));
+    }
+
+    [Fact]
+    public void CreateForEmptyDataUriThrows()
+    {
+        // Arrange
+        var data = BinaryData.Empty;
+
+        // Assert
+        Assert.Throws<ArgumentNullException>(() => new ImageContent(data));
+    }
+
+    [Fact]
+    public void ToStringForDataUriFromBytesReturnsDataUriString()
+    {
+        // Arrange
+        var bytes = System.Text.Encoding.UTF8.GetBytes("this is a test");
+        var data = BinaryData.FromBytes(bytes, "text/plain");
+        var content1 = new ImageContent(data);
+
+        // Act
+        var result1 = content1.ToString();
+        var dataUriToExpect = $"data:{data.MediaType};base64,{Convert.ToBase64String(data.ToArray())}";
+
+        // Assert
+        Assert.Equal(dataUriToExpect, result1);
+
+        // Assert throws if mediatype is null
+        Assert.Throws<ArgumentNullException>(() => new ImageContent(BinaryData.FromBytes(bytes, null)));
+        Assert.Throws<ArgumentNullException>(() => new ImageContent(BinaryData.FromBytes(bytes, string.Empty)));
+        Assert.Throws<ArgumentNullException>(() => new ImageContent(BinaryData.FromBytes(bytes, " ")));
+    }
+
+    [Fact]
+    public void ToStringForDataUriFromStreamReturnsDataUriString()
+    {
+        // Arrange
+        var ms = new System.IO.MemoryStream(System.Text.Encoding.UTF8.GetBytes("this is a test"));
+        try
```

Comment:
tests projects support c#11 you can do it simpler.
instead `System.Text.Encoding.UTF8.GetBytes("this is a test")`
you can write `"this is a test"u8

Reply:
Unfortunately the `MemoryStream` ctor does not take a `ROS<byte>` and `u8` gives back a `ROS<byte>`.

I could do `"this is a test"u8.ToArray()` not sure if is any better, though